src/sql/src/catalog.rs

```diff
@@ -872,13 +872,11 @@ pub enum CatalogType<T: TypeReference> {
 
 /// A description of a field in a [`CatalogType::Record`].
 #[derive(Clone, Debug, Eq, PartialEq)]
-pub struct CatalogRecordField<T: TypeReference> {
+pub struct CatalogRecordField {
     /// The name of the field.
     pub name: ColumnName,
-    /// The ID of the type of the field.
-    pub type_reference: T::Reference,
```

Comment:
All this generic stuff needs to stay. We just added some of this last week in https://github.com/MaterializeInc/materialize/pull/22082/files.

Reply:
Ah I saw these fields were used to eventually create a `ColumnType` in https://github.com/MaterializeInc/materialize/pull/22082/files#diff-28dcff29ce185a34dc7bb2444006600d245dd969d5bcd3a4e9bef65a36b5efc1R5382-R5388, which I needed to return the `RelationDesc` which is why I stored that directly.

Reply:
What I was missing here was the `ScalarType`.

Reply:
From the slack convo, let me see what it would look like without making type a relation. Will not need any modification here then.

Reply:
@mjibson From the updated convo, going the route of having a `desc` implementation for the composite type. So, this type reference contains the Global ID of the type. Is there another way to get a `ScalarType` and eventually a `ColumnType` from it without storing it here? Note, even if the `scalar_type` is stored here instead of the `column_type`, the `type_reference` is going to remain unused.

Reply:
Ah, the trick is to store the desc in the `CatalogItem` itself. That way you can avoid changing any of this code (which is delicate) and the `desc()` will still be able to return a `RelationDesc` without having access to the catalog.

Let me bear that theory out real quick.